File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string,vector<string>> mp;
        int n = strs.size();

        for(int i=0;i<n;i++){
            string ans = strs[i];
            sort(ans.begin(),ans.end());

            mp[ans].push_back(strs[i]);
        }

        for(auto x:mp){
            res.push_back(x.second);
        }

        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    from collections import defaultdict

    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        mp = defaultdict(list)

        for s in strs:
            key = ''.join(sorted(s))  # sorted string as key
            mp[key].append(s)

        return list(mp.values())